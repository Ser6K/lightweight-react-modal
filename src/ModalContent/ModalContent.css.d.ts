declare namespace ModalContentCssNamespace {
  export interface IModalContentCss {
    content: string;
  }
}

declare const ModalContentCssModule: ModalContentCssNamespace.IModalContentCss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: ModalContentCssNamespace.IModalContentCss;
};

export = ModalContentCssModule;
