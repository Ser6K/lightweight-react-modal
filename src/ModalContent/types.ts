import { ClassNameType, ReactNodeType } from 'src/types'

export interface ModalContentTypes {
  children: ReactNodeType
  className?: ClassNameType
  [propName: string]: any
}
