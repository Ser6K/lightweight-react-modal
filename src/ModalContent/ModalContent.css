.content {
  flex-grow: 1;
  overflow: auto;
  -webkit-overflow-scrolling: touch;
  height: auto;
  margin: 20px 15px;
}
