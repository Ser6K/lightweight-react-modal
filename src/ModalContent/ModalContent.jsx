import React from 'react';
import PropTypes from 'prop-types';
import { ContentDisplayName, classNames } from 'utils';
import styles from './ModalContent.css';

const ModalContent = ({ children, className, ...props }) => (
    <div
        {...props}
        className={classNames(styles.content, className)}
    >
        {children}
    </div>
);

ModalContent.propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
};

ModalContent.defaultProps = {
    className: null,
};

ModalContent.displayName = ContentDisplayName;
export default React.memo(ModalContent);
