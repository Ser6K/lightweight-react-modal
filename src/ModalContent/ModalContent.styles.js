export default {
    content: {
        flexGrow: 1,
        overflow: 'auto',
        '-webkit-overflow-scrolling': 'touch',
        height: 'auto',
        margin: '20px 15px',
    },
};
