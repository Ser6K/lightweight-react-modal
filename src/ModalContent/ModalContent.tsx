import React from 'react';
import { ContentDisplayName, classNames } from 'utils';
import styles from './ModalContent.css';

interface ModalContentProps {
    children: React.ReactNode,
    className?: string,
    [propName: string]: any,
};

const ModalContent = React.memo(({
    children,
    className,
    ...props
}: ModalContentProps) => (
    <div
        {...props}
        className={classNames(styles.content, className)}
    >
        {children}
    </div>
));

ModalContent.displayName = ContentDisplayName;
export default ModalContent;