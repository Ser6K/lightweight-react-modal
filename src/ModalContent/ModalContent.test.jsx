import React from 'react';
import { mount } from 'enzyme';
import ModalContent from './ModalContent';

describe('Modal content', () => {
    test('it matches snapshot with no props', () => {
        const modalContent = mount(<ModalContent>Content</ModalContent>);
        expect(modalContent).toMatchSnapshot();
    });
    test('it matches snapshot with props', () => {
        const modalContent = mount(<ModalContent className="className" style={{ display: 'inline' }}>Content</ModalContent>);
        expect(modalContent).toMatchSnapshot();
    });
});
