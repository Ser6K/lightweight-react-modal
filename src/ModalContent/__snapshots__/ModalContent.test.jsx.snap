// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal content it matches snapshot with no props 1`] = `
<ModalContent>
  <div
    className="content-0-2-1"
  >
    Content
  </div>
</ModalContent>
`;

exports[`Modal content it matches snapshot with props 1`] = `
<ModalContent
  className="className"
  style={
    Object {
      "display": "inline",
    }
  }
>
  <div
    className="content-0-2-2 className"
    style={
      Object {
        "display": "inline",
      }
    }
  >
    Content
  </div>
</ModalContent>
`;
